src/mcp/server/auth/middleware/bearer_auth.py

```diff
@@ -0,0 +1,100 @@
+"""
+Bearer token authentication middleware for ASGI applications.
+
+Corresponds to TypeScript file: src/server/auth/middleware/bearerAuth.ts
+"""
+
+import time
+from typing import List, Optional, Callable, Awaitable, cast, Dict, Any
+
+from starlette.requests import HTTPConnection, Request
+from starlette.exceptions import HTTPException
+from starlette.authentication import AuthCredentials, AuthenticationBackend, AuthenticationError, BaseUser, SimpleUser, UnauthenticatedUser, has_required_scope
+from starlette.middleware.authentication import AuthenticationMiddleware
+from starlette.types import Scope
+
+from mcp.server.auth.errors import InsufficientScopeError, InvalidTokenError, OAuthError
+from mcp.server.auth.provider import OAuthServerProvider
+from mcp.server.auth.types import AuthInfo
+
+
+class AuthenticatedUser(SimpleUser):
+    """User with authentication info."""
+    
+    def __init__(self, auth_info: AuthInfo):
+        super().__init__(auth_info.user_id or "anonymous")
+        self.auth_info = auth_info
+        self.scopes = auth_info.scopes
+
+
+class BearerAuthBackend(AuthenticationBackend):
+    """
+    Authentication backend that validates Bearer tokens.
+    """
+    
+    def __init__(
+        self,
+        provider: OAuthServerProvider,
+    ):
+        self.provider = provider
+    
+    async def authenticate(self, conn: HTTPConnection):
+
+        if "Authorization" not in conn.headers:
+            return None
+            
+        auth_header = conn.headers["Authorization"]
+        if not auth_header.startswith("Bearer "):
+            return None
+            
+        token = auth_header[7:]  # Remove "Bearer " prefix
```

Comment:
I think this can be simplified by
```suggestion
        auth_header = conn.headers.get("Authorization")
        if not auth_header or not auth_header.startswith("Bearer "):
            return None

        token = auth_header[7:]  # Remove "Bearer " prefix
```